### PR TITLE
Run tests & tslint for src/addons directory

### PR DESCRIPTION
**Pull request by vladimirze**
Long story short, I wanted to work on a bug but I noticed that tests and the tslint are not running for `src/addons` directory (intentionally?), this is my attempt to fix this 

- removed `src/addons` from `exclude` from tsconfig.js
- fixed tslint errors
- before the change there were 549 passing tests, now there are 560

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -20,8 +20,5 @@
   "include": [
     "src/**/*",
     "typings/xterm.d.ts"
-  ],
-  "exclude": [
```
Comment (Tyriar) [inline tsconfig.json]: We can't not exclude this as the source is part of a different tsconfig.json. The tests are meant to be included here:

https://github.com/xtermjs/xterm.js/blob/37972aa5dda107da569dded73863815bf25a0cea/gulpfile.js#L25-L30

`yarn test` runs `gulp mocha` which picks up these source files. Maybe there's a problem with the gulp task? Did you have the addons built before you ran yarn test (`yarn watch-addons`)?

Comment (vladimirze) [reply tsconfig.json]: To reproduce this, I've deleted `build/`, `lib/` directories and run `yarn build` and `yarn test`.
It seems that the issue is with only `src/search.test.js` that gets lost somewhere. I've installed gulp plugin to output the file paths that `gulp.src()` emits in the `mocha` task:

`
[22:03:00] Using gulpfile ~/src/xterm.js/gulpfile.js
[22:03:00] Starting 'mocha'...
[22:03:00] Using ./lib/Buffer.test.js
[22:03:00] Using ./lib/BufferLine.test.js
[22:03:00] Using ./lib/BufferSet.test.js
[22:03:00] Using ./lib/CharWidth.test.js
[22:03:00] Using ./lib/CompositionHelper.test.js
[22:03:00] Using ./lib/EscapeSequenceParser.test.js
[22:03:00] Using ./lib/InputHandler.test.js
[22:03:00] Using ./lib/Linkifier.test.js
[22:03:00] Using ./lib/SelectionManager.test.js
[22:03:00] Using ./lib/SelectionModel.test.js
[22:03:00] Using ./lib/Terminal.test.js
[22:03:00] Using ./lib/common/CircularList.test.js
[22:03:00] Using ./lib/common/EventEmitter.test.js
[22:03:00] Using ./lib/common/Lifecycle.test.js
[22:03:00] Using ./lib/public/Terminal.test.js
[22:03:00] Using ./lib/handlers/Clipboard.test.js
[22:03:00] Using ./lib/renderer/CharacterJoinerRegistry.test.js
[22:03:00] Using ./lib/renderer/ColorManager.test.js
[22:03:00] Using ./lib/renderer/GridCache.test.js
[22:03:00] Using ./lib/utils/Clone.test.js
[22:03:00] Using ./lib/utils/MouseHelper.test.js
[22:03:00] Using ./lib/utils/TestUtils.test.js
[22:03:00] Using ./lib/ui/CharMeasure.test.js
[22:03:00] Using ./lib/addons/fit/fit.test.js
[22:03:00] Using ./lib/addons/fullscreen/fullscreen.test.js
[22:03:00] Using ./lib/addons/attach/attach.test.js
[22:03:00] Using ./lib/addons/terminado/terminado.test.js
[22:03:00] Using ./lib/addons/winptyCompat/winptyCompat.test.js
[22:03:00] Using ./lib/addons/zmodem/zmodem.test.js
[22:03:00] Using ./lib/addons/webLinks/webLinks.test.js
[22:03:00] Using ./lib/core/input/Keyboard.test.js
[22:03:00] Using ./lib/renderer/atlas/LRUMap.test.js
[22:03:00] Using ./lib/renderer/dom/DomRendererRowFactory.test.js
[22:03:00] Using ./lib/Terminal.integration.js
`

File: src/addons/search/tsconfig.json [hunk 2]
```diff
@@ -18,8 +18,5 @@
   },
   "include": [
     "**/*.ts"
-  ],
-  "exclude": [
-    "**/*.test.ts"
```
Comment (Tyriar) [inline src/addons/search/tsconfig.json]: I could reproduce as well and it looks like this was the problem, Can you undo the change to `<root>/tsconfig.json`?

Comment (vladimirze) [reply src/addons/search/tsconfig.json]: I squashed the commits, the change in `<root>/tsconfig.json` is no longer there.
